frontend/src/features/codes/OwnersList.tsx

```diff
@@ -0,0 +1,113 @@
+    const [isModifyOwnerModalOpen, setIsModifyOwnerModalOpen] = useState(false);
+    const [owner, setOwner] = useState<IOwner | undefined>(undefined);
+
+    // action for the row click
+    const editFn = (ownerEdited: IOwner) => {
+        setIsModifyOwnerModalOpen(true);
+        setOwner(ownerEdited);
+    };
+    return (
+        <QueryStateHandler
+            data={data}
+            error={error}
+            isLoading={isLoading}
+        >
+            <div className="list-headers">
+                <div>Nimi</div>
+                <div>Henkilö- tai Y-tunnus</div>
+            </div>
+            <ul className="results-list">
+                {data?.contents.map((owner: IOwner) => (
+                    <div
+                        key={owner.identifier}
+                        className="results-list__item"
+                        onClick={(e) => {
+                            e.preventDefault();
+                            editFn(owner);
+                        }}
+                    >
+                        <span>{owner.name}</span>
+                        <span>{owner.identifier}</span>
+                    </div>
+                ))}
+            </ul>
+            <span className="list-footer">
+                Näytetään {data?.page.size}/{data?.page.total_items} hakutulosta
+            </span>
+            <ModifyOwnerModal
+                owner={owner as IOwner}
+                isVisible={isModifyOwnerModalOpen}
+                setIsVisible={setIsModifyOwnerModalOpen}
+            />
+        </QueryStateHandler>
+    );
+};
+
+export default function OwnersList() {
+    // search strings
+    const [filterParams, setFilterParams] = useState<IFilterOwnersQuery>({name: "", identifier: ""});
+
+    // focus the field when clicking its cross circle icon
+    const nameSearchRef = useRef<HTMLInputElement>(null);
+    const identifierSearchRef = useRef<HTMLInputElement>(null);
+    const focus = useCallback(
+        (field: string) => {
+            if (field === "name") {
+                nameSearchRef.current?.focus();
+            } else if (field === "identifier") {
+                identifierSearchRef.current?.focus();
+            }
+        },
+        [nameSearchRef, identifierSearchRef]
+    );
+    const clearAndFocus = (field: string) => {
+        setFilterParams((prev) => ({...prev, [field]: ""}));
+        focus(field);
+    };
+
+    return (
+        <div className="listing">
+            <div className="filters">
+                <TextInput
+                    id="filter__name"
+                    ref={nameSearchRef}
+                    label="Nimi"
+                    value={filterParams.name}
+                    onChange={(e) => setFilterParams((prev) => ({...prev, name: e.target.value}))}
+                    onButtonClick={() => clearAndFocus("name")}
+                    autoFocus
+                    buttonIcon={filterParams.name ? <IconCrossCircle /> : <IconSearch />}
+                />
```

Comment:
Why not use `FilterTextInputField` here? 🤔

Reply:
The comment above applies to this too.